src/tron/tronPlugin.ts

```diff
@@ -27,11 +30,37 @@ export class TronPlugin extends CurrencyPlugin {
   }
 
   async importPrivateKey(userInput: string): Promise<Object> {
-    throw new Error('Must implement checkBlockchainInnerLoop')
+    if (/^(0x)?[0-9a-fA-F]{64}$/.test(userInput)) {
+      // It looks like a private key, so validate the hex:
+      const tronKeyBuffer = Buffer.from(userInput.replace(/^0x/, ''), 'hex')
+      if (EthereumUtil.isValidPrivate(tronKeyBuffer) === true) {
+        throw new Error('Invalid private key')
+      }
+      const tronKey = tronKeyBuffer.toString('hex')
+      return { tronKey }
+    } else {
+      // it looks like a mnemonic, so validate that way:
+      if (!validateMnemonic(userInput)) {
+        throw new Error('Invalid input')
+      }
+      const tronKey = await this._mnemonicToTronKey(userInput)
+      return {
+        tronMnemonic: userInput,
+        tronKey
+      }
+    }
   }
 
   async createPrivateKey(walletType: string): Promise<Object> {
-    throw new Error('Must implement checkBlockchainInnerLoop')
+    const type = walletType.replace('wallet:', '')
```

Comment:
This is wrong. You should be doing  `if (walletType !== this.currencyInfo.walletType) { throw ... }`, instead of slicing up the strings.